internal/component/database_observability/postgres/collector/explain_plan.go

```diff
@@ -417,7 +417,10 @@ func (c *ExplainPlan) fetchExplainPlans(ctx context.Context) error {
 			continue
 		}
 
-		if !strings.HasPrefix(strings.ToLower(qi.queryText), "select") {
+		containsReservedWord := database_observability.ContainsReservedKeywords(qi.queryText, database_observability.ExplainReservedWordDenyList)
+
+		if containsReservedWord {
+			level.Debug(logger).Log("msg", "skipping query containing reserved word", "query", qi.queryText)
```

Comment:
Could just the `queryid` be enough? Or if we want to log the query text explicitly maybe we should do the same on mysql?